lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -109,8 +148,13 @@ protected function run($argument): void {
 				->andWhere($qb->expr()->like('path', $qb->createNamedParameter($path . '%')))
 				->andWhere($qb->expr()->eq('storage', $qb->createNamedParameter($storageId)))
 				->andWhere($qb->expr()->in('mimetype', $qb->createNamedParameter($mimeTypes, IQueryBuilder::PARAM_INT_ARRAY)))
-				->andWhere($qb->expr()->gt('filecache.fileid', $qb->createNamedParameter($lastFileId)))
-				->orderBy('filecache.fileid', 'ASC')
+				->andWhere($qb->expr()->gt('filecache.fileid', $qb->createNamedParameter($lastFileId)));
+
+			foreach ($ignoreDir as $dir) {
+				$files = $files->andWhere($qb->expr()->neq('parent', $qb->createNamedParameter($dir['fileid'])));
+			}
+
+			$files = $files->orderBy('filecache.fileid', 'ASC')
```

Comment:
$files should be the IResult of the query as returned by executeQuery(), while all modifications to the query should be done to the query builder: $qb -- it's a bit confusing as I wrote it :)